spam/spam_tutorial.py

```diff
@@ -140,234 +145,406 @@
+# * **Develop labeling functions in isolation.** Avoid using a "boosting" approach, trying to develop LFs specifically to interact with behavior of existing LFs. If you use the existing LFs to identify uncovered areas of the data, generalize the observed pattern rather than writing an LF to cover those exact examples.
+# * **Prioritize accuracy over coverage, except when accuracy is very high.** Very accurate LFs often result in higher performance models than ones with more coverage. However, coverage is often more difficult to obtain. So choose a 90% accuracy and 30% coverage version of an LF over a 95% accuracy and 10% coverage version. But choose an 80% accuracy LF and 20% coverage version of an LF over a 65% accuracy and 40% coverage version.
+# * **Try to cover as many data points as possible, but avoid overfitting.** In order to extract the most value out of your training set, regularly identify areas of the training set that haven't been covered yet and create new LFs that cover them. Don't obssess over covering the entire data set, as LFs written to cover outliers won't generalize.
+# * **Aim to have overlaps.** We'll train a _label model_ to denoise and combine the outputs of our LFs. The label model learns from overlaps and conflicts between LFs, so it's helpful if multiple LFs cover certain areas of the data set.
 #
-# Once multiple LFs have been created, users can look at data points that have received no labels so far (or many conflicting labels) to get ideas for new LFs to write.
-# Furthermore, if there are some classes that have votes from very few LFs, users might iterate by writing additional LFs to cover those parts of the dataset.
-# **Note, however, that it is not necessary for LFs to assign labels to every data point;** in fact, most of the time your LFs will not have perfect dataset-wide coverage.
-# We rely on the fact that the classifier that trains on labels from Snorkel has the power to _generalize_ and can therefore learn a good representation of the data even if the each data point in the training set does not receive a label from any LFs.
-# In addition, note that **it is ok to have overlaps and conflicts between labeling functions**. For example, two good labeling functions can conflict when one of them is written to fix the errors of the other. We later use a _label model_ that learns how to combine the outputs of all (potentially conflicting) labeling functions to estimate label probabilities.
+# Of course, rules are meant to be broken.
+# For example, we can sometimes increase performance by adding LFs that label only if certain others don't.
+# And in some advanced problems, a 95% accurancy and 10% coverage LF may in fact be preferable to a 90% accurancy and 30% coverage one.
+# Use the principles above to guide your process, but don't treat them as hard-and-fast rules.
```

Comment:
I only think this confuses things more- especially coming before the reader has seen any LFs.  Would cut (or put it at the very end after they've seen an iteration loop)